pkg/server/slicerouter_config.go

```diff
@@ -257,14 +255,21 @@ func vl3ReconcileRoutesInKernel() error {
 	logger.GlobalLogger.Infof("Route map: %v", routeMap)
 	logger.GlobalLogger.Infof("Slice Route map: %v", remoteSubnetRouteMap)
 
-	for remoteSubnet, nextHop := range remoteSubnetRouteMap {
-		_, ok := routeMap[remoteSubnet]
-		// If the route is absent or the nexthop is incorrect, reinstall the route.
-		if !ok || routeMap[remoteSubnet].Gw.String() != nextHop {
-			logger.GlobalLogger.Infof("Installed route does not reflect slice state. Reconciling dst: %v, gw: %v", remoteSubnet, nextHop)
-			err := vl3InjectRouteInKernel(remoteSubnet, nextHop)
-			if err != nil {
-				logger.GlobalLogger.Errorf("Failed to install route: dst: %v, gw: %v", remoteSubnet, nextHop)
+	for remoteSubnet, nextHopIpList := range remoteSubnetRouteMap {
+		for i := 0; i < len(nextHopIpList); i++ {
+			_, ok := routeMap[remoteSubnet]
+			// If the route is absent or the nexthop is incorrect, reinstall the route.
+			if !ok || routeMap[remoteSubnet].Gw.String() != nextHopIpList[i] {
```

Comment:
With ecmp we will have multiple routes for the remoteSubnet. So shouldnt the routeMap have multiple (atleast two) entries for the remoteSubnet? This comparison does not look right.